### DISCUSSION TITLE
Setting require authentication on helm chart

### DISCUSSION THREAD

**Discussion by danielgauci** [datalust/seq-tickets#1575] [Ideas]
Hey guys,

Hoping that in future releases we can set `SEQ_FIRSTRUN_REQUIREAUTHENTICATIONFORHTTPINGESTION` or any other environment variables (maybe a helm value that is evaluated in a template within the chart/deployment.yaml) to set all available seq config variables listed at https://docs.datalust.co/docs/json-config.

Can this be done currently?

**Comment by KodrAus** [comment]
Hi @danielgauci :wave: The chart doesn't currently expose a way to set arbitrary environment variables, these particular ones under the `firstRun` configuration property might be best set in an [init script](https://docs.datalust.co/docs/init-scripts), but I think it would also be reasonable to add a `firstRun` property to your `Values.yaml` and read them from there.

**Reply by danielgauci** [reply]
Thanks for the quick response @KodrAus.

Could you provide an example on how to provide the init scripts to the helm chart? Inspection of the helm chart `deployment.yaml` file doesn't give any hints.

Alternatively if the `firstRun` property will be included in the helm chart in the short future we can wait for these properties.

It would be awesome that all config variables mentioned in https://docs.datalust.co/docs/json-config could be configured and not just the `FIRSTRUN_*` vars.

**Reply by KodrAus** [reply]
I think the most straightforward way to add an init script would be to drop it in the persistent volume you mount, but that's not always easy to get to.

One of the reasons we haven't just exposed all configuration to the chart directly is that some Seq configuration (particularly around the API) doesn't map one-to-one with Kubernetes concepts, but needs to be set to support it, so we configure it based on how you set up your ingress etc. Adding additional knobs that conflict with those would make support more difficult.

I think our current scheme is a little ad-hoc though. We actually do appear to have `firstRunAdminUsername` and `firstRunAdminPasswordHash` already, so I'll just add the other `firstRun` properties alongside them.

**Comment by KodrAus** [comment]
I've opened https://github.com/datalust/helm.datalust.co/pull/29 to add support for setting `firstRunRequireAuthenticationForHttpIngestion` to the Helm chart.

**Comment by KodrAus** [comment]
Now that https://github.com/datalust/helm.datalust.co/pull/29 is merged, this should make it into our next patch release, which I think we've got planned for the coming week.
